pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -611,6 +616,8 @@ private boolean processStreamEvents(MessageBatch messagesAndOffsets, long idlePi
       if (_segmentLogger.isDebugEnabled()) {
         _segmentLogger.debug("empty batch received - sleeping for {}ms", idlePipeSleepTimeMillis);
       }
+      // Record Pinot ingestion delay as zero since we are up-to-date and no new events
```

Comment:
At least at this point, we know we have no other events then we sleep.

Reply:
Discussed this in our online meet up: there should be a contract with Stream providers not to provide empty batch if there are events. We currently go to sleep when that happen and this will negatively affect performance for those streams that return empty and do have events. We also discussed options to handle this differently for each stream provider using a method in the stream provider that will indicate if zeroing is valid for the specific stream provider.

Reply:
@Jackie-Jiang to propose appropriate interface. I suggest an interface in the stream metadata like (just as an example) `emptyReturnIndicatesNoEventsAvailable()`. I think what Jackie had in mind was to change the poll API to indicate whether there are more events. That could be a return in the metadata, for example. I am open to either approach.

Reply:
I think the interface should be in the consumer as that is the one which is polling the source and knows the state of the upstream partition. I would like to point out that only the consumer (or subscriber) will know its own state that relative to its upstream. 

In general, I urge the team to think about solutions that can be cleanly extended/adapted to all other streaming systems.